device_backends/LogicalNameMapping/src/LogicalNameMapParser.cc

```diff
@@ -104,11 +104,12 @@ namespace ChimeraTK {
       T value;
       try {
         // base=0 means auto detect base, depending on prefix
-        long long longVal = std::stoll(valAsString, nullptr, 0);
+        int64_t longVal = std::stoll(valAsString, nullptr, 0);
         if(std::is_unsigned_v<T> && longVal < 0) {
           throw std::out_of_range(std::string("negative!"));
         }
-        if(longVal > (long long)std::numeric_limits<T>::max()) {
+        // we need to exclude upper range check for uint64 since max val does not fit into longVal
+        if(!std::is_same_v<T, uint64_t> && longVal > (int64_t)std::numeric_limits<T>::max()) {
```

Comment:
This is only checking positive overflow. To be complete there could be a check that negative values are not too small if T is signed.

Reply:
I now added an explicit check for the lower bound